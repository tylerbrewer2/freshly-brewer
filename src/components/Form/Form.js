import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

// Components
import Row from '../Row';
import Col from '../Col';
import Header from '../Header';
import Button from '../Button';
import Flash from '../Flash';

const FormCore = ({
  onSubmit,
  title,
  submitButtonText,
  children,
  mobileWidth,
  tabletWidth,
  requestResponse,
  className
}) => {
  function handleOnSubmit(e) {
    e.preventDefault();
    onSubmit(e);
  } 

  return (
    <Row center>
      <Col mobileWidth={mobileWidth} tabletWidth={tabletWidth}>
        <section className={className}>
          <Header marginBottom="15px">{title}</Header>
          <form onSubmit={handleOnSubmit}>
            {children}

            <Col mobileWidth={12}>
              <Button
                type='submit'
                fullWidth
              >
                {submitButtonText}
              </Button>
            </Col>
            {requestResponse &&
              <Flash status={requestResponse.status}>
                {requestResponse.message}
              </Flash>
            }
          </form>
        </section>
      </Col>
    </Row>
  )
}

FormCore.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  submitButtonText: PropTypes.string,
  mobileWidth: PropTypes.string.isRequired,
  tabletWidth: PropTypes.string.isRequired,
  requestResponse: PropTypes.shape({
    status: PropTypes.string,
    message: PropTypes.string,
  }),
}

FormCore.defaultProps = {
  submitButtonText: "SUBMIT",
  requestResponse: undefined,
}

const Form = styled(FormCore)`
  margin-top: 100px;
  margin-bottom: 100px;
`

export default Form;
