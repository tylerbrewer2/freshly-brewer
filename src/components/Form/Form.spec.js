import React from 'react';
import renderer from 'react-test-renderer';

import Form from './Form';

describe('component structure', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <Form
        onSubmit={jest.fn()}
        title="Test title"
        submitButtonText="Test message"
        requestResponse={{
          status: 'error',
          message: 'Test message',
        }}
      >
        Test child
      </Form>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
