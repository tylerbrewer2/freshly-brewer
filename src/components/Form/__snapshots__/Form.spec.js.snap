exports[`component structure renders correctly 1`] = `
<div
  className="undefined  center-xs row">
  <div
    className="col-xs-undefined 
        undefined 
        undefined">
    <section
      className="sc-bxivhb eianJZ">
      <h1
        className="sc-bdVaJa fOkMPb">
        Test title
      </h1>
      <form
        onSubmit={[Function]}>
        Test child
        <div
          className="col-xs-12 
              undefined 
              undefined">
          <button
            className="sc-bwzfXH bOFSPc"
            type="submit">
            Test message
          </button>
        </div>
        <section
          className="col-xs-12">
          <div
            className="sc-htpNat fuMuJW">
            Test message
          </div>
        </section>
      </form>
    </section>
  </div>
</div>
`;
